### PR TITLE
[JBPM-10016] Drools/jBPM integration: high number of instances waiting for signal adversely impacts execution time

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-10016

### CODE REVIEW

**Comment by elguardian** [comment]
Jenkins retest this please

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/AsyncIntermediateCatchSignalTest.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ public void testSyncGlobalSignal() {
         ksession.signalEvent("MySignal", null);
     }
 
-    @Test(timeout = 20000)
+    @Test//(timeout = 20000)
```
Comment (gmunozfe) [inline jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/AsyncIntermediateCatchSignalTest.java:139]: remove comment or increase timeout

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java [hunk 2]
```diff
@@ -69,6 +71,55 @@ public void setKnowledgeRuntime(InternalKnowledgeRuntime kruntime) {
...
+            return toLong((List<String>) value);
+        }
+        return Collections.emptyList();
+    }
+
+    private List<Long> toLong(List<String> items) {
+        List<Long> longList = new ArrayList<>();
+        for (String item : items) {
+            try {
+                longList.add(Long.parseLong((String) item));
```
Comment (gmunozfe) [inline jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java:114]: Cast to String not needed

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java [hunk 3]
```diff
@@ -69,6 +71,55 @@ public void setKnowledgeRuntime(InternalKnowledgeRuntime kruntime) {
         this.kruntime = kruntime;
     }
 
+    @Override
+    public Collection<ProcessInstance> loadKnowledgeRuntimeProcessInstances() {
+        InternalRuntimeManager manager = (InternalRuntimeManager) kruntime.getEnvironment().get(EnvironmentName.RUNTIME_MANAGER);
+        List<Long> ids = Collections.emptyList();
```
Comment (gmunozfe) [inline jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java:77]: ids will take a value if the "if or else" clause, so this assignment is useless.

**Review by gmunozfe — APPROVED** [review]
Looks good to me, awesome work @elguardian !
Some minor comments, nothing really important.

File: jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/AsyncIntermediateCatchSignalTest.java (hunk 1 above)
Comment (elguardian) [reply jbpm-test-coverage/src/test/java/org/jbpm/test/functional/async/AsyncIntermediateCatchSignalTest.java:139]: done

File: jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java (hunk 3 above)
Comment (elguardian) [reply jbpm-persistence/jbpm-persistence-jpa/src/main/java/org/jbpm/persistence/processinstance/JPAProcessInstanceManager.java:77]: done

**Pull request merged**